File: scripts/sdk/Conformance.java
import io.chronograph.Client;
import com.google.gson.*;
import java.io.*;
import java.nio.charset.StandardCharsets;
import java.time.Duration;
import java.util.HexFormat;
public class Conformance {
 public static void main(String[] args) throws Exception {var input=new BufferedReader(new InputStreamReader(System.in,StandardCharsets.UTF_8));String line;while((line=input.readLine())!=null){var r=new JsonObject();try{var q=JsonParser.parseString(line).getAsJsonObject();var c=new Client(q.get("url").getAsString(),q.get("token").getAsString(),Duration.ofMillis(q.has("timeout")?q.get("timeout").getAsLong():30000),q.has("limit")?q.get("limit").getAsInt():4194304);JsonElement v;if(q.has("construct"))v=new JsonPrimitive(true);else if(q.has("method"))v=new JsonPrimitive(HexFormat.of().formatHex(c.request(q.get("path").getAsString(),q.get("method").getAsString(),q.has("body")?q.getAsJsonObject("body"):null)));else v=c.call(q.get("op").getAsString(),q.has("body")?q.getAsJsonObject("body"):null);r.addProperty("ok",true);r.add("value",v);}catch(Client.ApiException e){r.addProperty("ok",false);r.addProperty("status",e.status);r.addProperty("code",e.code);r.addProperty("retry",e.retryAfter);}catch(Exception e){r.addProperty("ok",false);r.addProperty("local",true);r.addProperty("type",e.getClass().getSimpleName());}System.out.println(r);}}
}

File: sdk/dart/tool/conformance.dart
import 'dart:io';
import 'dart:convert';
import 'package:chronograph/chronograph.dart';

Future<void> main() async {
  await for (final line
      in stdin.transform(utf8.decoder).transform(const LineSplitter())) {
    ChronographClient? client;
    try {
      final q = jsonDecode(line) as Map<String, dynamic>;
      client = ChronographClient(q['url'], q['token'],
          timeout: Duration(milliseconds: q['timeout'] ?? 30000),
          maxResponseBytes: q['limit'] ?? 4194304);
      dynamic result;
      if (q['construct'] == true) {
        result = true;
      } else if (q['method'] != null) {
        final bytes = await client.request(q['path'],
            method: q['method'], body: q['body']);
        result = bytes.map((b) => b.toRadixString(16).padLeft(2, '0')).join();
      } else {
        result = await client.call(q['op'], q['body'] ?? {});
      }
      stdout.writeln(jsonEncode({'ok': true, 'value': result}));
    } on ApiException catch (e) {
      stdout.writeln(jsonEncode({
        'ok': false,
        'status': e.status,
        'code': e.code,
        'retry': e.retryAfter
      }));
    } catch (e) {
      stdout.writeln(jsonEncode(
          {'ok': false, 'local': true, 'type': e.runtimeType.toString()}));
    } finally {
      client?.close();
    }
  }
}
